ophyd/signal.py

```diff
@@ -129,20 +163,24 @@ def set_thread():
                              exc_info=ex)
                 success = False
             else:
+                logger.debug('set_and_wait(%r, %s) succeeded => %s', self.name,
+                             value, self.value)
                 success = True
                 if settle_time is not None:
                     time.sleep(settle_time)
             finally:
-                self._set_thread = None
                 st._finished(success=success)
+                self._set_thread = None
```

Comment:
Looks like I swapped lines unintentionally - is it worth bothering to swap them back?